javaagent/build.gradle.kts

```diff
@@ -216,6 +216,12 @@ tasks {
     dependsOn(shadowJar, baseJavaagentJar)
   }
 
+  if ((findProperty("removeJarVersionNumbers") ?: "false").toString().toBoolean()) {
```

Comment:
Nit
```suggestion
  if (findProperty("removeJarVersionNumbers") as Boolean) {
```

Reply:
This doesn't work actually... I get an NPE in CI and the following error locally:
```
class java.lang.String cannot be cast to class java.lang.Boolean (java.lang.String and java.lang.Boolean are in module java.base of loader 'bootstrap')
```

Reverting...

Reply:
We use this construct for testing whether `testLatestDeps` flag is set which works because of https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/aa2c2436c28919fa3d31cb426a08b7b99e983a6b/conventions/src/main/kotlin/io.opentelemetry.instrumentation.base.gradle.kts#L30-L31 Makes me wonder whether `findProperty("removeJarVersionNumbers") == "true"` would work

Reply:
Good call.  That did it.  I've updated both PRs.